versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/condition/MongoDBConditionAliasVisitor.java

```diff
@@ -0,0 +1,113 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl.condition;
+
+import java.util.List;
+
+import com.dremio.nessie.versioned.store.Entity;
+import com.google.common.collect.ImmutableList;
+
+/*
+ * This uses the Visitor design pattern to retrieve object attributes.
+ */
+public class MongoDBConditionAliasVisitor implements ConditionAliasVisitor {
+  /**
+   * The callback method in the visitor design pattern.
+   * @param conditionExpression The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return the aliased ConditionExpression.
+   */
+  @Override
+  public ConditionExpression visit(ConditionExpression conditionExpression, AliasCollector collector) {
+    return ImmutableConditionExpression.builder()
+      .functions(conditionExpression.getFunctions().stream()
+        .map(f -> f.acceptExpressionFunction(this, collector)).collect(ImmutableList.toImmutableList()))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param expressionFunction The object to be aliased.
+   * @param arguments constituent arguments of the object to be aliased.
+   * @param name the name of the object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return the aliased ExpressionFunction.
+   */
+  @Override
+  public ExpressionFunction visit(ExpressionFunction expressionFunction, List<Value> arguments,
+                                  ExpressionFunction.FunctionName name, AliasCollector collector) {
+    return new ExpressionFunction(name, arguments.stream().map(v -> getArgumentValue(v, collector))
+      .collect(ImmutableList.toImmutableList()));
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param expressionpath The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return The aliased ExpressionPath.
+   */
+  @Override
+  public ExpressionPath visit(ExpressionPath expressionpath, AliasCollector collector) {
+    return ImmutableExpressionPath.builder()
+      .root((ExpressionPath.NameSegment) expressionpath.getRoot().accept(this, collector))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param nameSegment The object to be aliased.
+   * @param collector the class that does the aliasing.
+   * @return The aliased ExpressionPath.
+   */
+  @Override
+  public ExpressionPath.NameSegment visit(ExpressionPath.NameSegment nameSegment, AliasCollector collector) {
+    return ImmutableNameSegment.builder()
+      .name(collector.escape(nameSegment.getName()))
+      .child(nameSegment.getChild().map(p -> p.alias(collector)))
+      .build();
+  }
+
+  /**
+   * The callback method in the visitor design pattern.
+   * @param value The object to be aliased.
+   * @param entity ValueOfEntity value
```

Comment:
Can you clarify this param comment, it's not informative to me.

Reply:
Yes, added clarification. We need ValueOfEntity (entity) rather than value. This is exposing private attributes. Again, if I expose those private attributes I need this method's interface will be simpler and less confusing.

Reply:
Refactored so this method is removed.